src/page-multi-facility/MultiFacilityErrorBoundary.tsx

```diff
@@ -0,0 +1,48 @@
+import { navigate } from "gatsby";
+import React, { useEffect } from "react";
+import { ErrorBoundary, FallbackProps } from "react-error-boundary";
+
+import { getBaselineScenario } from "../database";
+import Loading from "../design-system/Loading";
+import { useToasts } from "../design-system/Toast";
+import useScenario from "../scenario-context/useScenario";
+
+const MultiFacilityFallback: React.FC<FallbackProps> = ({
+  resetErrorBoundary,
+}) => {
+  const { addToast } = useToasts();
+  const [, dispatchScenarioUpdate] = useScenario();
+
+  useEffect(
+    () => {
+      getBaselineScenario().then((baselineScenario) => {
+        addToast(
+          "Something went wrong. You will be returned to your baseline scenario.",
+        );
+        if (baselineScenario) {
+          dispatchScenarioUpdate(baselineScenario);
+        }
+        navigate("/");
+        resetErrorBoundary();
```

Comment:
What happens if an error happens in the calls before this one within the `then` block? Are we left in some broken state where any page wrapped by this error boundary will fail because we didn't reset the boundary? I'm wondering if we should be doing something like a try-finally to ensure the reset is _always_ called.

Reply:
good question; I did some digging on this. 

If an error happens _inside_ the fallback error recovery page, we are entering catastrophic failure territory. As it currently stands, if we fail to execute the `navigate` call you'll just be looking at the loading spinner forever. We could also toast that error, but we can't do anything more useful than that unless we come up with a second fallback state that is less likely to raise an error. (For what it's worth, I think an error inside this effect is quite unlikely, but it's certainly not impossible, especially if you are suddenly unable to reach firestore for some reason.)

If we naively reset the error boundary without fully executing our fallback behavior, we'll just trigger an infinite error loop of some sort (of the original permissions error, and possibly also of whatever error is happening in the fallback component, if it happens every time).

The best solution in that case may just be to render something more useful than a loading spinner when we find ourselves unable to recover from error we've caught.

Reply:
Got it. It sounds like this may be a scenario that is so rare that we should not engineer for it straight-away. If you disagree and think we should create some fallback copy that we display indicating a "catastrophic" failure has occurred, feel free to pursue that. Otherwise, I think we can proceed?

Reply:
agreed, i'm not too worried about it at this point